### PR TITLE
Confluence token authentication

### PR DESCRIPTION
**Pull request by zhifeiyue**
Support token authentication for Confluence backend.

For Confluence Data Center or server editions, just use the personal access token:
`perceval confluence 'http://[YOUR_CONFLUENCE_SITE]/' -t [PERSONAL_ACCESS_TOKEN]`

For Confluence cloud:
`perceval confluence -u 'someone@example.com' -t [API_TOKEN] 'https://[YOUR_SITE].atlassian.net/wiki'`

### CODE REVIEW

File: releases/unreleased/add-token-auth-for-confluence.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+title: 'Add token authentication to the Confluence backend'
+category: added
+author: devpod.cn <support@devpod.cn>
+issue:
+notes: >
+    Add token authentication to the Confluence backend.
```
Comment (sduenas) [inline releases/unreleased/add-token-auth-for-confluence.yml:6]:
```suggestion
    Authentication in `confluence` backend is supported using
    personal access tokens. Confluence Data Center and server
    editions 7.9 and later can use personal access tokens without
    a username. For Confluence Cloud, username and token are required.
```

**Review by sduenas — CHANGES_REQUESTED** [review]
Thanks so much for your PR. Would you mind to address my comments?

File: perceval/backends/core/confluence.py [hunk 2]
```diff
@@ -69,14 +70,17 @@ class Confluence(Backend):
     CATEGORIES = [CATEGORY_HISTORICAL_CONTENT]
 
     def __init__(self, url, tag=None, archive=None, ssl_verify=True,
-                 spaces=None, max_contents=MAX_CONTENTS):
+                 spaces=None, max_contents=MAX_CONTENTS,
+                 user=None, api_token=None):
```
[the remark this answers is not in the document]
Comment (zhifeiyue) [reply perceval/backends/core/confluence.py:74]: ok, I will add the docstring and increase the version number.

File: perceval/backends/core/confluence.py [hunk 3]
```diff
@@ -395,8 +402,21 @@ class ConfluenceClient(HttpClient):
     VHISTORICAL = 'historical'
 
     def __init__(self, base_url, archive=None, from_archive=False, ssl_verify=True,
-                 spaces=None, max_contents=MAX_CONTENTS):
-        super().__init__(base_url.rstrip('/'), archive=archive, from_archive=from_archive, ssl_verify=ssl_verify)
+                 spaces=None, max_contents=MAX_CONTENTS,
+                 user=None, api_token=None):
```
[the remark this answers is not in the document]
Comment (zhifeiyue) [reply perceval/backends/core/confluence.py:406]: Ok. Thanks for your review, I will update the docstring today.

**Comment by zhifeiyue** [comment]
> It looks good to me but, before we merge it, would you mind to squash the commits into a single one and sign it?

ok, I just finished it.

**Review by sduenas — APPROVED** [review]
LGTM

**Comment by sduenas** [comment]
Thanks for your contribution!

**Pull request merged**